### ISSUE TITLE
.props files are broken

### ISSUE THREAD

**Issue by inosik** [fsprojects/Paket#1233]
Since v2.24.7 framework specific `.props` files (as in xunit.core, or our beloved xunit.runner.visualstudio) do not work anymore.

The problem is, that the framework specific part of the path (`__paket__Package_Name_targets` MSBuild property) isn't set, when the file gets imported.

**Comment by inosik** [comment]
As I see now, the normal `.props` files are also broken, because we resolve the whole file name.
```xml
<Choose>
  <When Condition="$(TargetFrameworkIdentifier) == '.NETFramework' And $(TargetFrameworkVersion) == 'v4.5'">
    <PropertyGroup>
      <__paket__Package_Name_props>Package.Name</__paket__Package_Name_props>
    </PropertyGroup>
  </When>
</Choose>
```

**Comment by forki** [comment]
I can reproduce.

**Comment by forki** [comment]
could you please try latest alpha?

**Comment by inosik** [comment]
Thanks, that solved the issue. But now we have #1219 again.

@theggelund Are there any issues at all if the `.props` aren't imported at the top of the file?

**Comment by forki** [comment]
We have still #1219 covered for props that are global. There is an
integration test

**Comment by forki** [comment]
we can't put this up to the top since the properties for the case block are not defined at the top.

**Comment by theggelund** [comment]
Ordering is important. Using conditional checks for variable not set is used to support amongst other features, multi-target builds

Accordning to nuget, props should be put at the top and targets should be put at the bottom. 
Ordering is important, and by using conditionals that check if variable is set and giving a default value, multi-target solutions can be supported in visual studio

**Comment by inosik** [comment]
@forki Sorry to bother you with this again, but we have to reopen this issue. Framework specific `.props` files get imported at the top of the project again.

**Comment by forki** [comment]
do we already have an integration test for this? (and did I break it ?)

Can you name a public nuget package which has this issue?

**Comment by inosik** [comment]
The infamous xunit.runner.visualstudio strikes back again. I prepared [this repro sample](https://github.com/fsprojects/Paket/files/166733/repro.zip).

IIRC we have an integration test for the opposite case: `.props` files which reside directly in `build/`, not in `build/{framework}/`.

**Comment by forki** [comment]
I will take a look

**Comment by forki** [comment]
xUnitTests.expected.csproj is that what you expect it to be? In 28116e0 the test suggests that this is actually the result that paket gives you.

**Comment by forki** [comment]
so what did I do wrong in the test?

**Comment by inosik** [comment]
Sorry, somehow I edited the expected result back again to the wrong one. [This](https://github.com/fsprojects/Paket/files/166985/repro-correct.zip) is how I'd expect it to be, the `<Import />` should come either at the end of the file, or right after the `<Choose />`.

**Comment by forki** [comment]
I don't understand. Why should it go to the end?

**Comment by forki** [comment]
ah I see. the property is not defined

**Comment by forki** [comment]
so this was changed in #1487 and I'm not completely sure what the correct way is

**Comment by forki** [comment]
so I guess this one is "fixed", but please help with #1487
